docs/spruce-skill/messages.md

```diff
@@ -20,8 +20,146 @@ const message = await ctx.sb.message(locationId: UUID4, userId: UUID4, message:
 });
 ```
 
+### Push Notifications
+
+The example message above can be received by the user as a push notification via the Spruce mobile app. Without any further customization, the user will receive just the `message` string in the notification.
+
+If you want to customize the push notification, simply add a `push` object to your message payload.
+
+```js
+// Basic notification
+const payload = {
+    push: {
+        title: String,// optional
+        subtitle: String,// optional
+        body: String// optional, will override `message` argument
+    }
+}
+```
+
+Additionally, Spruce takes advantage of "Rich Notification" features on iOS.
+
+```js
+// Send an image, brief video clip or audio clip
+// Media notification example:
+const mediaPushPayload = {
+    push: {
+        title: "VIP Alert",
+        body: "Alex Martinez just arrived at 1234 Main St",
+        media: {
+            url: "https://images.unsplash.com/photo-1501625277806-e25bd4596da3",
+            fileName: "1501625277806.png"
+        }
+    }
+}
+
+// Send a notification with customizable "actions"
+const actionsPushPayload = {
+    push: {
+        title: "This has some actions",
+        body: "Just press on this notification to see the actions",
+        actions: [
+            {
+                id: "action_one",
+                label: "Action One",
+                deepLink: "link_to_app_screen",// optional
+                webUrl: "link_to_web"// optional, lower priority than `deepLink`
+            },
+            {
+                id: "action_two",
+                label: "Action Two",
+                deepLink: "link_to_app_screen",// optional
+                webUrl: "link_to_web"// optional, lower priority than `deepLink`
+            }
+        ]
+    }
+}
+
+// Send a message as a user in an "iMessage" style UI
+const messagePushPayload = {
+    push: {
+        title: "Lindsay Brockling",
+        body: "Thanks so much again for the great service today. I can't wait for my next visit!",
+        message: {
+            user: {
+                id: '00000001',
+                name: 'Lindsay Brockling',
+                profileImages: {
+                    "profile60@2x": "https://images.unsplash.com/photo-1489424731084-a5d8b219a5bb"
+                }
+            },
+            message: "Thanks so much again for the great service today. I can't wait for my next visit!"
+        }
+        
+    }
+}
+
+// Send a Card as a notification:
+const cardPushPayload = {
+    push: {
+        title: "Card Notification",
+        card: {
+            cardBuilder: {
+                __typename: "CardBuilder",
```

Comment:
Are these `__typename` necessary?

Reply:
For now, yes they are necessary. Eventually I'd like to build out a solution that doesn't need this, but for now I need it to transform the json into the native GraphQL type.